File: framework/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/model/PromotionsWidget.java
package com.sdl.ecommerce.dxa.model;

import com.sdl.ecommerce.api.model.Promotion;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;

import java.util.List;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * Promotions Widget
 *
 * @author nic
 */
@SemanticEntity(entityName = "PromotionsWidget", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class PromotionsWidget extends AbstractEntityModel {

    @SemanticProperty("e:category")
    private ECommerceCategoryReference categoryReference;

    @SemanticProperty("e:product")
    private ECommerceProductReference productReference;

    @SemanticProperty("e:viewType")
    private String viewType;

    @SemanticProperty("e:maxPromotions")
    private Integer maxPromotions;

    private List<Promotion> promotions;

    public ECommerceCategoryReference getCategoryReference() {
        return categoryReference;
    }

    public ECommerceProductReference getProductReference() {
        return productReference;
    }

    public String getViewType() {
        return viewType;
    }

    public Integer getMaxPromotions() {
        return maxPromotions;
    }

    public List<Promotion> getPromotions() {
        return promotions;
    }

    public void setPromotions(List<Promotion> promotions) {
        this.promotions = promotions;
    }

    public String getViewName(Promotion promotion) {
        return promotion.getClass().getInterfaces()[0].getSimpleName();
    }

}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/PromotionsWidget.cs
using Sdl.Web.Common.Models;
using SDL.ECommerce.Api.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SDL.ECommerce.DXA.Models
{
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "PromotionsWidget", Prefix = "e")]
    public class PromotionsWidget : EntityModel
    {
        [SemanticProperty("e:category")]
        public ECommerceCategoryReference CategoryReference { get; set; }

        [SemanticProperty("e:product")]
        public ECommerceProductReference ProductReference { get; set; }

        [SemanticProperty("e:viewType")]
        public string ViewType { get; set; }

        [SemanticProperty("e:maxPromotions")]
        public int? MaxPromotions { get; set; }

        [SemanticProperty(IgnoreMapping = true)]
        public IList<IPromotion> Promotions { get; set; }
    }
}